### PR TITLE
Add Priority Class setting

### PR DESCRIPTION
**Pull request by ttpcodes**
This PR adds a new setting to `Longhorn` that allows a user to specify a `Priority Class` for the various `Longhorn` workloads to use as requested in longhorn/longhorn#1487. This allows users to set a higher priority for `Longhorn` workloads and can prevent the workloads from being the first to get evicted in the case of a `Node` pressure situation, such as memory pressure.

Just like with the `Taint Toleration` setting, this `Setting` change will cause all the `Longhorn` components to get redeployed once changed. As such, this setting is placed under `Danger Zone` with a warning to not modify the `Setting` while there are attached `Volumes`.

### CODE REVIEW

**Comment by shuo-wu** [comment]
Do we need to verify if the PriorityClass exists or not before updating the setting?

**Comment by ttpcodes** [comment]
On further investigating, I found out that `Pods` will fail to be created if the specified `Priority Class` does not exist, and figured that it would be better to verify the existence of the `Priority Class` first before updating the `Setting`. There is now a check to make sure the `Priority Class` is either `empty` or exists before updating the `Setting`.

File: datastore/longhorn.go [hunk 1]
```diff
@@ -109,6 +109,22 @@ func (s *DataStore) ValidateSetting(name, value string) (err error) {
 				return fmt.Errorf("cannot modify toleration setting before all volumes are detached")
 			}
 		}
+	case types.SettingNamePriorityClass:
+		if value != "" {
+			_, err := s.GetPriorityClass(value)
+			if err != nil {
```
Comment (shuo-wu) [inline datastore/longhorn.go:115]: `if _, err := s.GetPriorityClass(value); err != nil {`

**Comment by ttpcodes** [comment]
Fixed the typo I made with not condensing the `err != nil` check.

File: datastore/datastore.go [hunk 2]
```diff
@@ -112,6 +117,8 @@ func NewDataStore(
 		pvcStoreSynced: persistentVolumeClaimInformer.Informer().HasSynced,
 		knLister:       kubeNodeInformer.Lister(),
 		knStoreSynced:  kubeNodeInformer.Informer().HasSynced,
+		pcLister:       priorityClassInformer.Lister(),
+		pcStoreSynced:  priorityClassInformer.Informer().HasSynced,
```
Comment (shuo-wu) [inline datastore/datastore.go:121]: Add this to `func (s *DataStore) Sync(stopCh <-chan struct{}) bool`.

File: deploy/install/01-prerequisite/02-serviceaccount.yaml [hunk 3]
```diff
@@ -27,6 +27,9 @@ rules:
 - apiGroups: ["batch"]
   resources: ["jobs", "cronjobs"]
   verbs: ["*"]
+- apiGroups: ["scheduling.k8s.io"]
+  resources: ["priorityclasses"]
```
Comment (shuo-wu) [inline deploy/install/01-prerequisite/02-serviceaccount.yaml:31]: Do we need to update `longhorn-uninstall-role` in `longhorn-manager/deploy/uninstall/uninstall.yaml`?

Comment (ttpcodes) [reply deploy/install/01-prerequisite/02-serviceaccount.yaml:31]: I don't think so. The `Uninstall Controller` doesn't do anything to the `Priority Classes` during the uninstall process, and doesn't even monitor `Priority Classes` as part of its logic.

**Comment by yasker** [comment]
@joshimoo Can you help to review the PR too? Thanks.

**Pull request merged**